src/test/apisix/templates/configmap.yaml

```diff
@@ -0,0 +1,159 @@
+      #     - 9200
+      #     - 9211
+      # dns_resolver:
+      #   {{- range $resolver := .Values.dns.resolvers }}
+      #   - {{ $resolver }}
+      #   {{- end }}
+      dns_resolver_valid: {{.Values.dns.validity}}
+      resolver_timeout: {{.Values.dns.timeout}}
+      ssl:
+        enable: {{ .Values.gateway.tls.enabled }}
+        enable_http2: {{ .Values.gateway.tls.http2.enabled }}
+        listen_port: {{ .Values.gateway.tls.containerPort }}
+        ssl_protocols: "TLSv1 TLSv1.1 TLSv1.2 TLSv1.3"
+        ssl_ciphers: "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA"
+
+    nginx_config:                     # config for render the template to genarate nginx.conf
+      error_log: "/dev/stderr"
+      error_log_level: "warn"         # warn,error
+      worker_processes: auto
+      worker_rlimit_nofile: 20480     # the number of files a worker process can open, should be larger than worker_connections
+      worker_shutdown_timeout: 240s   # timeout for a graceful shutdown of worker 
+      event:
+        worker_connections: 10620
+      http:
+        access_log: "/dev/stdout"
+        keepalive_timeout: 60s         # timeout during which a keep-alive client connection will stay open on the server side.
+        client_header_timeout: 60s     # timeout for reading client request header, then 408 (Request Time-out) error is returned to the client
+        client_body_timeout: 60s       # timeout for reading client request body, then 408 (Request Time-out) error is returned to the client
+        send_timeout: 10s              # timeout for transmitting a response to the client.then the connection is closed
+        underscores_in_headers: "on"   # default enables the use of underscores in client request header fields
+        real_ip_header: "X-Real-IP"    # http://nginx.org/en/docs/http/ngx_http_realip_module.html#real_ip_header
+        real_ip_from:                  # http://nginx.org/en/docs/http/ngx_http_realip_module.html#set_real_ip_from
+          - 127.0.0.1
+          - 'unix:'
+        #lua_shared_dicts:              # add custom shared cache to nginx.conf
+        #  ipc_shared_dict: 100m        # custom shared cache, format: `cache-key: cache-size`
+
+    etcd:
+      host:
+      {{- if .Values.etcd.enabled }}
+        - "http://{{ include "apisix.fullname" . }}-etcd-0.{{ include "apisix.fullname" . }}-etcd.{{ .Release.Namespace }}.svc.{{ .Values.gateway.k8s_domain }}:2379"
```

Comment:
this url doesn't work when I deployed it with release name `api6`:

```
root@i-05aw7poe:/tmp/helm-charts/src/test/apisix# kubectl -n apps logs api6-apisix-67df64b6b4-cbwxk
/usr/local/openresty/luajit/bin/luajit: /usr/bin/apisix:831: curl http://api6-apisix-etcd-0.api6-apisix-etcd.apps.svc.cluster.local:2379/v2/keys/apisix/routes?prev_exist=false -X PUT -d dir=true --connect-timeout 3 --max-time 6 --retry 1 2>&1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: api6-apisix-etcd-0.api6-apisix-etcd.apps.svc.cluster.local
Warning: Transient problem: timeout Will retry in 1 seconds. 1 retries left.
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: api6-apisix-etcd-0.api6-apisix-etcd.apps.svc.cluster.local

stack traceback:
	[C]: in function 'error'
	/usr/bin/apisix:831: in function </usr/bin/apisix:770>
	/usr/bin/apisix:922: in main chunk
	[C]: at 0x562faa0cf320
```

Reply:
let me check

Reply:
the main reason here is due to the moment apisix started, while the etcd domain is not ready in DNS server.
so  it should need some efforts from apisix side. WIP.

Reply:
I add  something below to detect  the etcd domain is ready or not,
```
  initContainers:
      - name: init-apisix
        image: busybox:1.28
        command: ['sh', '-c', "until nslookup {{ .Release.Name }}-etcd-0.{{ .Release.Name }}-etcd.{{ .Release.Namespace }}.svc.{{ .Values.gateway.k8s_domain }}; do echo waiting for etcd; sleep 2; done"]
```
@hlwanghl what do you think?